### ISSUE TITLE
Related content: conflict with .Site.Params.keywords

### ISSUE THREAD

**Issue by jmooring** [gohugoio/hugo#9188]
With this in the site configuration:

```toml
[params]
keywords = ['a']
```

And this in layouts/_default/single.html

```go-html-template
{{ $related := .Site.RegularPages.Related . | first 5 }}
```

This error is thrown:

> Error: Error building site: failed to render pages: render of "page" failed: "/home/jmooring/code/hugo-testing/layouts/_default/single.html:4:22": execute of template failed: template: _default/single.html:4:22: executing "main" at <.Site.RegularPages.Related>: error calling Related: indexing currently not supported for index "keywords" and type []interface {}

Tested with v0.89.4 and v0.54.0 -- this isn't a new issue.

Minimal reproducible example:

```bash
git clone --single-branch -b hugo-forum-topic-35657 https://github.com/jmooring/hugo-testing hugo-forum-topic-35657
cd hugo-forum-topic-35657
hugo server
```

Then edit the site configuration, changing the name of the `keywords` key under `[params]` to `xxx`, and run `hugo` again... no problems.

Reference: <https://discourse.gohugo.io/t/using-regularpages-related-gives-me-keywords-index-support-error/35657/>

**Comment by Base-IQ-ly-Loud** [comment]
Wait. My phone's dying.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by Base-IQ-ly-Loud** [comment]
Gonna shower then come through - phone's on 1percent.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by ssube** [comment]
I started running into this as well, after updating a very old site to build on the v0.90 family. I tried the change proposed in the first post, renaming `.Params.tags` (in my case) to `.Params.xxx` seemed to fix the issue, _until_ I updated a layout to use the new name as well. Switching the type of `.Params.tags` from a dict to a list does not seem to make a difference, and I tried wrapping any references to `.Params.tags` in both `{{ if .Params.tags }}` and `{{ with .Params.tags }}`, with no luck.

In my case, the `.Params.tags` block was used by the theme to style some elements, but I had redefined most of those CSS rules in a static file, and was able to remove the block entirely. I'm not sure that will be helpful for everyone.

More specifically, I have a `layouts/_default/single.html` adapted from a/the minimal theme, which contains this template:

```
{{ $related := first 3 (.Site.RegularPages.Related .) }}

{{ if $related }}
    <h4 class="page-header">Related</h4>
    {{ range $related }} {{ partial "list-item" . }} {{ end }}
{{ end }}
```

And `layouts/_default/list-item.html` that contains this template:

```
    {{ if .Params.languages }}
    {{ range .Params.languages }}
    <a href="/languages/{{ . | urlize }}"><kbd class="tag-language">{{ . }}</kbd></a>
    {{ end }}
    {{ end }}
    {{ if .Params.platforms }}
    {{ range .Params.platforms }}
    <a href="/platforms/{{ . | urlize }}"><kbd class="tag-platform">{{ . }}</kbd></a>
    {{ end }}
    {{ end }}
    {{ if .Params.tags }}
    {{ range .Params.tags }}
    <a href="/tags/{{ . | urlize }}"><kbd class="tag-item">{{ . }}</kbd></a>
    {{ end }}
    {{ end }}
```

With this in my config:

```
 taxonomies:
   language: languages
   platform: platforms
   tag: tags

params:
  tags:
    item: '#283593'
    language: '#2E7D32'
    platform: '#6A1B9A'
    type: '#FF8F00'
```

And received an error for each time the `tags` were used:

```
ERROR 2022/05/18 17:39:31 render of "page" failed: "/home/ssube/code/github/ssube/robots-rip/layouts/_default/single.html:10:33": execute of template failed: template: _default/single.html:10:33: executing "_default/single.html" at <.Site.RegularPages.Related>: error calling Related: indexing currently not supported for index "tags" and type map[string]interface {}
ERROR 2022/05/18 17:39:31 render of "page" failed: "/home/ssube/code/github/ssube/robots-rip/layouts/_default/single.html:10:33": execute of template failed: template: _default/single.html:10:33: executing "_default/single.html" at <.Site.RegularPages.Related>: error calling Related: indexing currently not supported for index "tags" and type map[string]interface {}
```

Renaming `params.tags` to something that `layouts/default/_single.html` does not use did temporarily fix the error, because (in my case) the `{{ range ... }}` was never reached or run. Updating the template broke it again.

**Comment by jmooring** [comment]
This was fixed in v0.110.0 with https://github.com/gohugoio/hugo/commit/d595419031b247d6f15a615f6e70b1956ff8eab1.
